library/psa_crypto.c

```diff
@@ -1379,7 +1379,13 @@ psa_status_t psa_hash_setup( psa_hash_operation_t *operation,
                              psa_algorithm_t alg )
 {
     int ret;
-    operation->alg = 0;
+
+    /* A context must be freshly initialized before it can be set up. */
+    if (operation->alg != 0)
```

Comment:
Style (here and in a few other places):
```suggestion
    if ( operation->alg != 0 )
```